conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -940,38 +948,101 @@ public Boolean visitUnknown(String unknownType) {
     private CollectionType getCollectionType(Type type) {
         return type.accept(new DefaultTypeVisitor<>() {
             @Override
-            public CollectionType visitList(ListType _value) {
-                return new CollectionType(ConjureCollectionType.LIST, options.nonNullCollections());
+            public CollectionType visitList(ListType value) {
+                if (!options.nonNullCollections() || !options.primitiveOptimizedCollections()) {
+                    return new CollectionType(
+                            ConjureCollectionType.LIST, ConjureCollectionNullHandlingMode.NULLABLE_COLLECTION_FACTORY);
+                }
```

Comment:
ah true 3490ad8f

Reply:
added slightly different codegen in this PR
https://github.com/palantir/conjure-java/pull/2306/files#diff-af480f5a1b91af7a42236746b4697dc2bb4f932e3d3dbafb8e6032d1d6efe0b8R567-R576

Reply:
I can put this change behind the feature flag and not remove newSet() and newList()

Reply:
583b0e07

Reply:
I think you pushed to your fork. If you have two remotes, you can remove your fork and rename the other remote to the name you're most familiar with

```bash
git remote remove myforkname
git remote rename palantirremotename myforkname
```
So in my case, it usually takes this form because I usually call my forked-source remote `upstream` and my the remote I push to `origin`:

```bash
git remote remove origin
git remote rename upstream origin
```

Reply:
ah yeah i did, oops